controller/src/main/java/io/pravega/controller/server/ControllerServiceStarter.java

```diff
@@ -286,14 +305,23 @@ protected void startUp() {
             // Wait for controller event processors to start.
             if (serviceConfig.getEventProcessorConfig().isPresent()) {
                 log.info("Awaiting start of controller event processors");
-                controllerEventProcessors.awaitRunning();
+                // if store client has failed because of session expiration, there are two possibilities where 
+                // controllerEventProcessors.awaitRunning may be stuck forever -
+                // 1. stream creation is retried indefinitely and cannot complete because of zk session expiration
+                // 2. event writer after stream creation throws exception. 
+                // In both of above cases controllerEventProcessors.startAsync may not get called. 
+                CompletableFuture.anyOf(storeClientFailureFuture, 
```

Comment:
We can have a scenario where `storeClientFailureFuture` completes (exceptionally) and `controllerEventProcessors.awaitRunning()` is invoked right?

Reply:
yes, it is possible that we have the awaitRunning blocked while storeClientFailureFuture is failed. This will fail the ControllerServiceStarter's startup whereby throwing an exception. 
This will cause ControllerServiceMain to thorw the exception and exit the process.